### DISCUSSION TITLE
Vectors vs Tensors

### DISCUSSION THREAD

**Discussion by AV25242** [mrdbourke/pytorch-deep-learning#890] [General]
I have been reading a lot of vector databases and RAG. After going through tensors I am a bit confused. So does vector database actually store a tensor ? Tensor is a multidimensional vector in a way.  Or if they are different then do vector databases support tensor ?

In a RAG application when would you use a Tensor ? Any pointers ?

**Comment by macbrenn** [comment]
Vectors are Tensors of Order 1. A great visual source for better understanding Tensors is the content produced by 3Blue1Brown. IMHO the underlying math provides provides great insight into the inner workings of ML.

To use RAG you create embeddings for some supplementary or private data (documents) to improve the accuracy of the LLM text generation process. This additional data is embedded and stored in a vector database. Subsequently, when you generate your prompt for the LLM, you first search your vector database for relevant material and then you include your search results in the prompt to the LLM.
